BrowserKit/Sources/ToolbarKit/LocationView.swift

```diff
@@ -153,13 +189,20 @@ class LocationView: UIView, UITextFieldDelegate, ThemeApplicable {
 
     // MARK: - UITextFieldDelegate
     public func textFieldDidBeginEditing(_ textField: UITextField) {
-        gradientLayer.frame = CGRect()
+        textField.text = urlAbsolutePath
+        updateGradientLayerFrame()
+        if !doesURLTextFieldExceedViewWidth {
+            animateURLText(textField, options: .transitionFlipFromRight, textAlignment: .natural)
+        }
         locationViewDelegate?.locationViewDidBeginEditing(textField.text?.lowercased() ?? "")
     }
 
     public func textFieldDidEndEditing(_ textField: UITextField) {
+        formatAndTruncateURLTextField()
         updateGradientLayerFrame()
-        applyTruncationStyleToURLTextField()
+        if !doesURLTextFieldExceedViewWidth, !urlTextField.isFirstResponder {
+            animateURLText(textField, options: .transitionFlipFromLeft, textAlignment: .center)
```

Comment:
Did you test that the gradient works correctly with a right-to-left language as well?